score-vmware/src/main/java/io/cloudslang/content/vmware/services/helpers/MorObjectHandler.java

```diff
@@ -0,0 +1,44 @@
+package io.cloudslang.content.vmware.services.helpers;
+
+import com.vmware.vim25.*;
+import io.cloudslang.content.vmware.connection.ConnectionResources;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by Mihai Tusa.
+ * 3/22/2016.
+ */
+public class MorObjectHandler {
+    public ManagedObjectReference getVmMor(ConnectionResources connectionResources, String filter, String parameter) throws Exception {
+        ManagedObjectReference reference = connectionResources.getMorRootFolder();
+        return getSpecificMor(connectionResources, reference, filter, parameter);
+    }
+
+    public Map<String, ManagedObjectReference> getSpecificObjectsMap(ConnectionResources connectionResources, String objectType)
+            throws InvalidPropertyFaultMsg, RuntimeFaultFaultMsg {
+        return connectionResources.getGetMOREF().inContainerByType(connectionResources.getMorRootFolder(), objectType);
+    }
+
+    public ManagedObjectReference getEnvironmentBrowser(ConnectionResources connectionResources, String filter)
```

Comment:
add 'parameter' to method signature, and change 'getProperty' method call to (connectionResources, reference, filter, parameter)